File: OOP_Tasks/OOP_Task6/Nurse.cpp
#pragma once
#include <iostream>

#include <string>
#include "Nurse.h"


Nurse::Nurse(int id, string name, string address, float salary, int overTime) 
	:Employee(id, name, address, salary)
{
	this->overTime = overTime;
}

Nurse::Nurse() :Employee()
{
	overTime = 0;
}

void Nurse:: setOverTime(int overTime)
{
	this->overTime = overTime;
}

int Nurse:: getOverTime()
{
	return (overTime);
}

float Nurse:: addOverTime()
{
	return (salary += (overTime * 500));
}

void Nurse:: printData()
{
	cout << "Nurse id: " << id <<
		"\tNurse name: " << name <<
		"\tNurse address: " << address <<
		"\tOverTime hours: " << overTime <<
		"\tNurse salary: " << addOverTime();
}

File: OOP_Tasks/OOP_Task3/Nurse.cs
class Nurse:Employee
{
    float overTime;
    
    public Nurse()
    {
        
    }
    
    public Nurse(int id, string name, string address, float salary, float overTime):base(id, name, address, salary)
    {
        this.overTime = overTime;
    }
    
    public float over_time
    {
        set { overTime = value; }
        get { return(overTime); }
    }

    // add 500 on each hour
    public float additionSalary()
    {
        return (salary += (500 * overTime));
    }
    
    public override string printData()
    {
        return ("nurse id: " + id + "\n" +
                "nurse name: " + name + "\n" +
                "nurse address: " + address + "\n" +
                "nurse salary: " + additionSalary() + "\n");
        
    }
}